homeassistant/components/econet/water_heater.py

```diff
@@ -66,7 +66,7 @@ class EcoNetWaterHeater(EcoNetEntity, WaterHeaterEntity):
     _attr_should_poll = True  # Override False default from EcoNetEntity
     _attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
 
-    def __init__(self, water_heater):
+    def __init__(self, water_heater: Equipment) -> None:
```

Comment:
```suggestion
    def __init__(self, water_heater: WaterHeater) -> None:
```
And this will be a WaterHeater.